plugin/src/main/kotlin/org/gradleweaver/plugins/jlink/JLinkOptions.kt

```diff
@@ -1,12 +1,22 @@
 package org.gradleweaver.plugins.jlink
 
+import org.gradle.api.Action
+import java.io.File
+
 class JLinkOptions(val name: String) {
 
     /**
-     * The modules to link. These MUST be on the module path or included in the JDK. If not set, `jdeps` will be run
-     * on the output JAR file from [shadowTask] to automatically determine the modules used.
+     * The application JAR. This will be copied into the generated jlink image's `bin` directory, next to the
+     * `java` executable.
+     */
+    var applicationJar: File? = null
```

Comment:
This should be a `RegularFileProperty`.

Reply:
`RegularFileProperty` or just `Provider<File>`? `RegularFileProperty` is difficult to bind to normal `File` objects like those provided by the outputs of the `Jar` task

Reply:
You want `RegularFileProperty`. I've fixed the `Jar` task for Gradle 5.1 to support being able to produce a `RegularFileProperty` directly.

Reply:
I'd prefer not to have to depend on bleeding edge releases. It limits the usability of the plugin

Reply:
Regardless, we should still just use the `RegularFileProperty`.

Reply:
`RegularFileProperty` is impossible to use in 4.10 when setting it to a normal `File `object like `Jar::archivePath` or `ShadowJar::archivePath`. And as I said before, I don't want to depend on a bleeding edge release and limit the usability of the plugin